Tests/SwiftDiagnosticsTest/NewDiagnosticsFormatterTests.swift

```diff
@@ -0,0 +1,222 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftDiagnostics
+import XCTest
+import SwiftSyntax
+import SwiftSyntaxBuilder
+import _SwiftSyntaxTestSupport
+import SwiftParser
+
+struct DiagnosticDescriptor {
+  let locationMarker: String
+  let id: MessageID
+  let message: String
+  let severity: DiagnosticSeverity
+  let highlight: [Syntax] // TODO: How to create an abstract model for this?
+  let noteDescriptors: [NoteDescriptor]
+  let fixIts: [FixIt] // TODO: How to create an abstract model for this?
```

Comment:
@DougGregor in #2166 mentioned:
> Note: a similar issue exists with Fix-Its, where we'd need a way to render what they actually do.

Do we already have a way to render this somewhere?

Reply:
I don't think we should try to render Fix-Its into the diagnostic output. We tried with the diagnostic formatter that we've been using in the compiler for years now, and it's hard to provide that much information via a console interface without obscuring the rest of the error. Fix-Its are best handled by editors.